fulp_modules/features/exclusive_fauna/exiled_king_basic.dm

```diff
@@ -0,0 +1,582 @@
+	charge_ability.Grant(src)
+	ai_controller.blackboard[BB_CHARGE_ABILITY] = WEAKREF(charge_ability)
+	QDEL_IN(src, 5 SECONDS)
+
+/datum/action/cooldown/mob_cooldown/charge/vanguard
+	charge_damage = 20
+	charge_delay = 0.7 SECONDS
+
+/datum/ai_controller/basic_controller/vanguard
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic(),
+	)
+
+	ai_movement = /datum/ai_movement/basic_avoidance
+	idle_behavior = /datum/idle_behavior/idle_random_walk/less_walking
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/ram_target,
+	)
+
+/datum/ai_planning_subtree/ram_target
+	var/datum/ai_behavior/ram/ram_behavior = /datum/ai_behavior/ram
+
+/datum/ai_planning_subtree/ram_target/SelectBehaviors(datum/ai_controller/controller, delta_time)
+	. = ..()
+	var/datum/weakref/weak_target = controller.blackboard[BB_BASIC_MOB_CURRENT_TARGET]
+	var/atom/target = weak_target?.resolve()
+	if(!target || QDELETED(target))
+		return
+	controller.queue_behavior(ram_behavior, BB_BASIC_MOB_CURRENT_TARGET, BB_TARGETTING_DATUM, BB_BASIC_MOB_CURRENT_TARGET_HIDING_LOCATION)
+	return SUBTREE_RETURN_FINISH_PLANNING
+
+/datum/ai_behavior/ram
+	action_cooldown = 0.6 SECONDS
+	behavior_flags =  AI_BEHAVIOR_MOVE_AND_PERFORM
+	required_distance = 15
+
+/datum/ai_behavior/ram/setup(datum/ai_controller/controller, target_key, targetting_datum_key, hiding_location_key)
+	. = ..()
+	var/datum/weakref/weak_target = controller.blackboard[target_key]
+	var/atom/target = weak_target?.resolve()
+	if(!target)
+		return FALSE
+
+/datum/ai_behavior/ram/perform(delta_time, datum/ai_controller/controller, target_key, targetting_datum_key, hiding_location_key)
+	. = ..()
+	var/mob/living/basic/basic_mob = controller.pawn
+	var/datum/weakref/weak_target = controller.blackboard[target_key]
+	var/atom/target = weak_target?.resolve()
+	var/datum/targetting_datum/targetting_datum = controller.blackboard[targetting_datum_key]
+
+	if(!targetting_datum.can_attack(basic_mob, target))
+		finish_action(controller, FALSE, target_key)
+		return
+
+	var/atom/hiding_target = targetting_datum.find_hidden_mobs(basic_mob, target) //If this is valid, theyre hidden in something!
+	var/atom/final_target = hiding_target ? hiding_target : target
+
+	if(!can_see(basic_mob, final_target, required_distance))
+		finish_action(controller, FALSE, target_key)
+		return
+
+	controller.blackboard[hiding_location_key] = WEAKREF(hiding_target)
+
+	var/datum/weakref/ability_weakref = controller.blackboard[BB_CHARGE_ABILITY]
+	var/datum/action/cooldown/mob_cooldown/charge_ability = ability_weakref?.resolve()
+	if(isnull(charge_ability))
+		return
+
+	charge_ability.Activate(final_target)
+
+/datum/ai_behavior/basic_ranged_attack/finish_action(datum/ai_controller/controller, succeeded, target_key, targetting_datum_key, hiding_location_key)
+	. = ..()
+	if(!succeeded)
+		controller.blackboard -= target_key
+		return
+
+/obj/item/clothing/neck/cloak/squid
```

Comment:
wasn't cloak armor not allowed for some reason? I think people are able to make themselves 100% immune with it.